core/src/validator.rs

```diff
@@ -346,6 +349,8 @@ impl ValidatorConfig {
             enforce_ulimit_nofile: false,
             rpc_config: JsonRpcConfig::default_for_test(),
             block_production_method: BlockProductionMethod::ThreadLocalMultiIterator,
+            replay_forks_threads: NonZeroUsize::new(8).expect("8 is non-zero"),
+            replay_transactions_threads: NonZeroUsize::new(8).expect("8 is non-zero"),
```

Comment:
This seemed like a reasonable value for tests. Only concern is maybe something like bench-tps that might use this `default_for_test()` method. I could search around, and update callers to do:
```rust
let config = ValidatorConfig {
    replay_transactions_threads: NonZeroUsize::new(num_cpus::num_cpus()).expect("num_cpus is non-zero")
     ..ValidatorConfig::default_for_tests()
};

Reply:
The prior default would result in using `get_max_thread_count`, which was equivalent to `num_cpus`, right?

I'm hesitant to modify the behavior as a stray part of this PR. Maybe as a separate follow-up where we can add some data on any changes to test run time, debate if we should have any num_cpu caps, etc.

Reply:
Yeah, good point ... better to stick with what we had. I'll use `get_max_thread_count()` just so it is very obvious that no behavior has changed; will clean up that custom rayon limiting crate at the end in one separate PR